### ISSUE TITLE
port mach_jet problem from Maestro

### ISSUE THREAD

**Issue by zingale** [AMReX-Astro/MAESTROeX#56]
We have no examples with inflow / user-specified boundary conditions.  We should port the mach_jet problem over to demonstrate / implement this functionality.

**Comment by harpolea** [comment]
I've looked into this again and think I've worked out what stumped me before. 

- In MAESTRO, the user-specified boundary conditions are specified in `mach_jet/multifab_physbc.f90`. This iterates over the components, calculating different boundary conditions depending on the component.
- In MAESTROex, the physical boundaries are defined and filled here https://github.com/AMReX-Astro/MAESTROeX/blob/4dee98c3414d0611b8644f2f5733b70d860ba667/Source/MaestroFillData.cpp#L92-L98
- The function that fills the boundaries is `phifill`, defined in `bc_fill_nd.F90`. This function *does not* take the component as an argument - it receives a 3d array representing a single component. 

There does not seem to be an obvious way to make `phifill` aware of the index of the component. `InterpFromCoarseLevel` takes this as an argument, but between that and the bit where AMReX actually calls `phifill` I'm not entirely sure what happens to it.
